cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -345,55 +343,62 @@ static void sc_mount_nvidia_driver_biarch(const char *rootfs_dir)
 
 #ifdef NVIDIA_MULTIARCH
 
-struct sc_nvidia_driver {
-	int major_version;
-	int minor_version;
-};
+typedef struct {
+	int major;
+	// Driver version format is MAJOR.MINOR[.MICRO] but we only care about the
+	// major version and the full version string. The micro component has been
+	// seen with relevant leading zeros (e.g. "440.48.02").
+	char raw[sizeof("1234.5678.90AB")];
```

Comment:
maybe we should use `sizeof("1234.567.90A\n") +1` to make it close to the actual version string

Reply:
but sizeof "" is not strlen "". This is doing the right thing.

Reply:
Ah, I see the `\n`! Totally right!